File: DownloadFiles/TGNet/Chapter08/VB/Exercise8_3/MaxOrdersTest/MaxOrdersTest/MaxOrdersTest/Form1.vb
Imports System.Data.Oledb
Public Class Form1

    Private Sub cmdADO_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdADO.Click
        Dim cn As New OleDbConnection _
             ("Provider=Microsoft.jet.oledb.4.0;Data Source=Northwind.mdb")
        Dim dr As OleDbDataReader
        Dim intCount As Int16 = 0

        cn.Open()
        Dim cmd As New OleDbCommand
        cmd.CommandText = _
         "SELECT Orderid, Orderdate FROM Orders " & _
                  "WHERE orderdate=(SELECT MAX(orderdate) FROM orders)"
        cmd.Connection = cn
        dr = cmd.ExecuteReader
        If dr.HasRows Then
            Do While dr.Read
                lstResults.Items.Add(dr.Item("OrderID").ToString + " " + _
                    dr.Item("Orderdate").ToString)
                intCount += 1
            Loop
            MessageBox.Show(intCount.ToString & " rows qualified")
        Else
            MessageBox.Show("no rows returned")
        End If
        dr.Close()
        cn.Close()

    End Sub
End Class

File: DownloadFiles/TGNet/Chapter08/C#/Exercise8-3/MaxOrdersTestAccess/MaxOrdersTest/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Data.OleDb;

namespace MaxOrdersTest
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void cmdADO_Click(object sender, EventArgs e)
        {
            OleDbConnection cn = new OleDbConnection("Provider=Microsoft.jet.oledb.4.0;Data Source=Northwind.mdb");
            OleDbDataReader dr;
            Int16 intCount = 0;
            cn.Open();
            OleDbCommand cmd = new OleDbCommand();
            cmd.CommandText = "SELECT Orderid, Orderdate FROM Orders " +
                 "WHERE orderdate = (SELECT MAX(orderdate) FROM orders)";
            cmd.Connection = cn;
            dr = cmd.ExecuteReader();
            if (dr.HasRows)
            {
                while (dr.Read())
                {
                    lstResults.Items.Add
                       (dr.GetInt32(dr.GetOrdinal("OrderID"))
                             + " " + dr.GetDateTime(dr.GetOrdinal("Orderdate")));
                    intCount += 1;
                }
                MessageBox.Show(intCount + " rows qualified");
            }
            else
            {
                MessageBox.Show("no rows returned");
            }
            dr.Close();
            cn.Close();

        }
    }
}